### PR TITLE
BBBIA-29 implemented disabling navigation buttons

### PR DESCRIPTION
**Pull request by anton-sarudko**

### CODE REVIEW

**Review by DevMile — APPROVED** [review]

File: BigBlueButton-RLP/Extensions/CGFloat+.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  CGFloat+.swift
+//  BigBlueButton-RLP
+//
+//  Created by Anton Sarudko on 10.12.21.
+//
+
+import UIKit
+
+extension CGFloat {
+    static let enabled = 1.0
```
[the remark this answers is not in the document]
Comment (anton-sarudko) [reply BigBlueButton-RLP/Extensions/CGFloat+.swift:11]: I will extract the values into an opacity enum

File: BigBlueButton-RLP/View/WebNavigationView.swift [hunk 2]
```diff
@@ -52,6 +52,16 @@ class WebNavigationView: UIView {
         fatalError("init(coder:) has not been implemented")
     }
     
+    func update(canGoBack: Bool, canGoForward: Bool) {
+        btnInteraction(backButton, isEnabled: canGoBack)
+        btnInteraction(forwardButton, isEnabled: canGoForward)
+    }
+    
+    private func btnInteraction(_ btn: UIButton, isEnabled: Bool) {
```
[the remark this answers is not in the document]
Comment (anton-sarudko) [reply BigBlueButton-RLP/View/WebNavigationView.swift:60]: I will change it to: interact(with btn: UIButton, isEnabled: Bool)

File: BigBlueButton-RLP/Scenes/WebViewController.swift [hunk 3]
```diff
@@ -113,6 +116,17 @@ class WebViewController: UIViewController, WKUIDelegate {
         //            }
         //        }
     }
+    
+    // Needed because didStartProvisionalNavigation and decidePolicyForNavigationAction don't fire if only part of the webpage is reloaded
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
```
[the remark this answers is not in the document]
Comment (anton-sarudko) [reply BigBlueButton-RLP/Scenes/WebViewController.swift:121]: Do you mean the colon after WebViewController ?
This is generated by Apple.

[the remark this answers is not in the document]
Comment (kwlz) [reply BigBlueButton-RLP/Scenes/WebViewController.swift:121]: `[NSKeyValueChangeKey: Any]?`

**Pull request merged**